### PR TITLE
return 0 in case maxPoints is null in the database

### PR DESCRIPTION
**Pull request by Olimbia23**
### Motivation and Context

Fix sentry issue [ARTEMIS-11X](https://sentry.ase.in.tum.de/organizations/artemis/issues/1099/?query=is%3Aunresolved+assigned_or_suggested%3Ame&statsPeriod=14d)

My solution: check if maxPoints is null and if that's the case return 0

### CODE REVIEW

**Comment by lschlesinger** [comment]
Question: do you know, why this occurs? And could you reproduce the error? I am wondering in which situations `this.maxPoints` can be undefined? Is it reasonable to call this method in the first place if the points are null? Maybe the fix could also be to disallow calling `calculateExamScores` in such situations...

File: src/main/java/de/tum/in/www1/artemis/domain/exam/Exam.java [hunk 1]
```diff
@@ -226,7 +226,7 @@ public void setConfirmationEndText(String confirmationEndText) {
     }
 
     public Integer getMaxPoints() {
```
Comment (krusche) [inline src/main/java/de/tum/in/www1/artemis/domain/exam/Exam.java:228]:
```suggestion
    public int getMaxPoints() {
```

File: src/main/java/de/tum/in/www1/artemis/service/exam/ExamService.java [hunk 2]
```diff
@@ -368,11 +368,6 @@ public void validateForStudentExamGeneration(Exam exam) throws BadRequestAlertEx
             }
         }
 
-        // Check that the exam max points is set
-        if (exam.getMaxPoints() == null) {
-            throw new BadRequestAlertException("The exam max points is not set.", "Exam", "artemisApp.exam.validation.maxPointsNotSet");
-        }
-
```
Comment (Olimbia23) [inline src/main/java/de/tum/in/www1/artemis/service/exam/ExamService.java:375]: Since now `exam.getMaxPoints` returns `int`, this check is redundant.

**Comment by Olimbia23** [comment]
> Question: do you know, why this occurs? And could you reproduce the error? I am wondering in which situations `this.maxPoints` can be undefined? Is it reasonable to call this method in the first place if the points are null? Maybe the fix could also be to disallow calling `calculateExamScores` in such situations...

The crash happens when one wants to access the scores of an exam, that has no maximum points defined.

Screenshot 2021-05-22 at 8 07 51 PM

Should we allow creation of an exam without maximum points or is it a bug, @krusche ?

File: src/main/java/de/tum/in/www1/artemis/service/exam/ExamService.java (hunk 2 above)
Comment (krusche) [reply src/main/java/de/tum/in/www1/artemis/service/exam/ExamService.java:375]: I think max points cannot be 0, so you should check that the value is not 0 now

**Comment by krusche** [comment]
It probably makes sense to make sure that maxPoints has a value > 0 when the instructor creates the exam, because a couple of other features (e.g. grade scales, and some validations later in the process) depend on it. In case the instructor does not know it, the value could also be changed afterwards.

**Review by lschlesinger — APPROVED** [review]
I tested to create an exam, validation works and the code looks good to me as well :)

**Pull request merged**